src/main/resources/db/migration/V1__init.sql

```diff
@@ -0,0 +1,33 @@
+create table scenario_results
+(
+    id          bigserial primary key,
+    user_id     bigint      not null,
+    name        text        not null,
+    site        text        not null,
+    executed_at timestamptz not null default now()
+);
+
+create table steps
+(
+    id     bigserial primary key,
+    action text not null,
+    value  text not null
+);
+
+create table step_results
+(
+    id        bigserial primary key,
+    step_id   bigint references steps (id) on delete cascade,
+    is_passed boolean
+);
+
+create table scenario_results_step_results
+(
+    scenario_result_id bigint not null,
+    step_result_id     bigint not null,
+    primary key (scenario_result_id, step_result_id),
+    constraint scenario_step_scenario_fk foreign key (scenario_result_id)
+        references scenario_results (id) on delete cascade,
+    constraint scenario_step_step_fk foreign key (step_result_id)
+        references step_results (id) on delete cascade
```

Comment:
I think it better setting foreign keys after initialization in all tables.

Reply:
Setting foreign keys after tables are initialized is a valid approach that can have certain advantages, especially when creating a database schema. It allows more flexible control over the order in which tables are initialized and can make data imports or migrations easier.

However, there are a few important points to consider:
1. Approach to migrations: If you plan to use database migrations such as Flyway or Liquibase, setting foreign keys should be part of the migration scripts. This ensures that the database schema is created and modified consistently according to your specification.
2. Data Security: Setting foreign keys after tables have been initialized can increase the risk of dirty data (data that does not comply with foreign key constraints) if data was inserted before the constraints were set. This can lead to undesirable consequences and difficulties in the future.
3. Data Dependencies: If your database design involves circular dependencies between tables, this may require more careful management of the order in which foreign keys are installed. In some cases, it may be necessary to temporarily disable or break foreign keys to insert initial data.

In general, when to install foreign keys depends on the specific requirements of your application and database methodology. Unless you have a specific reason for setting foreign keys after tables are initialized, it is generally recommended to include them in table definitions at table creation to ensure data integrity from the start.